test/extended/gssapi.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/bash
+#
+# Extended tests for logging in using GSSAPI
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+cd "${OS_ROOT}"
+source hack/lib/init.sh
+
+os::log::stacktrace::install
+trap os::test::junit::reconcile_output EXIT
+os::test::junit::declare_suite_start "test-extended/gssapiproxy"
+os::util::environment::setup_time_vars
+os::build::setup_env
+
+hack/build-go.sh cmd/oc -tags=gssapi
+os::cmd::expect_success_and_text 'oc version' 'GSSAPI Kerberos SPNEGO'
+
+function cleanup() {
+    out=$?
+    cleanup_openshift
+    echo "[INFO] Exiting"
+    oc project gssapiproxy
+    return $out
+}
+
+function wait_for_auth_proxy() {
+    local server_config=${1}
+    local spec='{.items[0].spec.containers[0].env[?(@.name=="SERVER")].value}_{.items[0].status.conditions[?(@.type=="Ready")].status}'
+    os::cmd::try_until_text "oc get pods -l deploymentconfig=gssapiproxy-server -o jsonpath='${spec}'" "^${server_config}_True$"
+}
+
+trap "exit" INT TERM
+trap "cleanup" EXIT
```

Comment:
You can't trap the `cleanup` and the `os::test::junit::reconcile_output` on EXIT at the same time. Call `os::test::junit::reconcile_output` at the end of `cleanup` to get that functionality